code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,193 @@
+/**
+	Time agent
+
+	You've got 5 minutes of playtime to do something weird
```

Comment:
I may have overlooked it, but it looks like you have no yet implemented the fail state/time limit extraction